pkg/manager/manager.go

```diff
@@ -164,6 +167,8 @@ func (k *KubeMacPoolManager) waitForKubevirt() {
 
 // wait for the any interrupt to stop the manager and clean the webhook.
 func (k *KubeMacPoolManager) waitForSignal() {
+	defer k.cancel()
```

Comment:
waitForSignal will be alive until one of the conditions is met isn't it ?
(because its a select without a default case)
If so, this code is prone to create some dangling go routines
maybe better to close them ?
since the loop 
https://github.com/k8snetworkplumbingwg/kubemacpool/pull/292/files#diff-d500fbd6a2aa620607ca5e2a7c3ac4f1a4c82309d1a549561e92abfcb18f2f0eR84
creates them

sorry if i am wrong

Reply:
I didn't find any such things. This is my reasoning:
- `waitForSignal` is supposed to be alive as long as kubemacpool is properly working, so normally there's nothing dangling. 

that leaves the two cases:
- In the case of kubevirt is pending then found by `checkForKubevirt`, then the `waitForKubevirt` goroutine will signal `waitForSignal` and close first, then `waitForSignal` goroutine will close, initiating the defer that will only then initiate the restart of the loop, and the goroutines will be recreated.. didn't find dangling goroutines here.. 

- the other case is that KMP gets a stop signal interrupt. in this scenario, assuming that kubevirt is already up, we only have `waitForSignal` running. then the signal notified to `waitForSignal` which breaks the loop and closes. didn't find dangling goroutines here as well.. 

IIUC, in the remote chance that both cases happen simultaniously, then we might have a dangling `waitForKubevirt` goroutine, but that's highly unlikely.. 

what do you think?

Reply:
Sorry i am not familiar with it enough
better that we discuss offline if possible, meanwhile lets not block the PR
thanks

Reply:
I think Or has a point in case `Start` returns an error. IIUC, the [docs](https://github.com/kubernetes-sigs/controller-runtime/blob/0d84ce43073cefb813563dd7586c943fa2c97783/pkg/manager/manager.go#L86) say that Start returns in two cases:
- an error occured
- context was canceled

If `Start` returns an error, then the goroutines are still running. If `Start` returns error in each iteration for some reason, then I think we may be spawning new goroutines in each iteration.
But this behaviour is not introduced by this PR

Reply:
ahh I see what you mean.. I will open an issue about it, to fix later, as indeed it's not introduced in this PR

Thanks @oshoval @rhrazdil for clarifying :)

Reply:
https://github.com/k8snetworkplumbingwg/kubemacpool/issues/293